### PR TITLE
Update DG UI component and setWindow

### LINKED ISSUE #259: Update DG UI component and setWindow
The UI component and `setWindow` (both implementation and test cases) have not yet been updated in the DG.

### PR DESCRIPTION
**Pull request by petermonky**
Closes #259.

### This PR:

* Updates the UI component and `setWindow` command sections of the DG

### Things to consider:

* UI component section does not elaborate on help window and reminder window
  * May have to rename reminder pane to something else to keep naming consistent; will also make documentation easier to write
* Design considerations section not included with `setWindow`; not sure if necessary

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -316,6 +318,28 @@ The following sequence diagram shows how the reminder feature works:
...
+
+#### Implementation <a id="implementation-set-reminder-window"></a>
+
+![UML diagram of the SetWindow feature](images/SetReminderWindowDiagram.png)
+
+Given below is an example usage scenario and how the set reminder window feature behaves at each step:
+1. The user executes the command `setWindow 14` to set the reminder window to 14 days.
+2. Then the `SetReminderWindowCommandParser#parse()` creates an instance of `SetReminderWindowCommand` by passing the number of days to which the reminder window will be set.
+4. The `SetReminderWindowCommand#execute()` method will update the `model` using the `Model#setReminderWindow()` method, which consequently updates `model.userPrefs` by calling the `UserPrefs#setReminderWindow()` method to update the reminder window as stored in the user preferences.
+6. The `Parser` returns the `CommandResult` which is then executed by `LogicManager`.
```
Comment (BrendonLau) [inline docs/DeveloperGuide.md]: MInor fix, should be 5.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -316,6 +318,28 @@ The following sequence diagram shows how the reminder feature works:
...
+The `setWindow` feature allows users to set the reminder window to a specific number of days. Any roles stored in the company list that fall within the set window will be displayed in the reminder pane that automatically opens up on start.
+
+#### Implementation <a id="implementation-set-reminder-window"></a>
+
+![UML diagram of the SetWindow feature](images/SetReminderWindowDiagram.png)
+
+Given below is an example usage scenario and how the set reminder window feature behaves at each step:
+1. The user executes the command `setWindow 14` to set the reminder window to 14 days.
+2. Then the `SetReminderWindowCommandParser#parse()` creates an instance of `SetReminderWindowCommand` by passing the number of days to which the reminder window will be set.
+4. The `SetReminderWindowCommand#execute()` method will update the `model` using the `Model#setReminderWindow()` method, which consequently updates `model.userPrefs` by calling the `UserPrefs#setReminderWindow()` method to update the reminder window as stored in the user preferences.
```
Comment (BrendonLau) [inline docs/DeveloperGuide.md]: is model.userPrefs referring to an userPref object in model?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (petermonky) [reply docs/DeveloperGuide.md]: Nice catch, I'll fix this right away 👍 thanks!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (petermonky) [reply docs/DeveloperGuide.md]: Yes that's right, I wasn't exactly sure how to express that so I just used `model.userPrefs`. Would simply `userPrefs` suffice?

Comment (BrendonLau) [reply docs/DeveloperGuide.md]: yup I think so.

Comment (petermonky) [reply docs/DeveloperGuide.md]: Okay fixed 👍 thanks!

**Review by kerbysoh — APPROVED** [review]
LGTM!

**Pull request merged**